### PR TITLE
Improve the resolving of email addresses given contact names

### LINKED ISSUE #73: Move utility functions to util folder
Utility functions connected to the general things we use such as contacts and email should be made reusable by moving the functions to a util folder. For example `Send.is_email(str)` can be useful in other modules.

### LINKED ISSUE #74: Improve the handling of meeting participants in the schedule module
The `to` field is currently just presumed to be a list of emails in order for the user to just enter names we have to be able to extract the correct email for each specified meeting participant (if it is not an email).

In the beginning this can just be implemented by using fuzzy to determine the correct email corresponding to the name.

### PR DESCRIPTION
**Pull request by hugowangler**
# Proposed changes
### Misc
* Created a `utils/` directory and moved existing `is_email()` from Schedule module to this folder to allow for reusability
* Updated the email Regex to also consider emails written in uppercase to be valid (had an issue with this and thought my code was incorrect for a while :P)

### Resolving emails
- Created a new util function which is able to resolve the email addresses of local contacts given their names (or valid emails directly). This is done by returning a dictionary containing:
    - emails: the known emails resolved (if input is a valid email they will automatically be added to emails)
   - uncertain: the name and potential candidates and their emails, will contain any contacts which could not be resolved i.e. multiple with same name
    - unknown: the input names which could not be resolved to any local
contacts

This function is now used when trying to resolve the emails of meeting attendees. Necessary follow-ups has also been added such as which contanct to pick (if uncertain is not empty) and if the meeting should still be scheduled without the unknown contact.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

- [x] "schedule a meeting with hello@email.com at 15:00" should create the meeting with the specified email without any follow-ups.
- [x] "schedule a meeting with Hugo at 15:00" should create the meeting and Hugo should be resolved to hugo@email.com so the guest in the calendar should be hugo@email.com
- [x] "schedule a meeting with John at 15:00" should prompt the user to pick which John to invite to the meeting. If the user responds with an invalid input (out of range or NaN) they should be prompted again until valid. On valid input the correct John should be listed as a guest in the calendar.
- [x] "schedule a meeting with Mark at 15:00" should prompt the user if the meeting should be scheduled without the unknown contact Mark.
- [x] `pytest` will run the tests I added for the utility function itself `get_emails()`

I have not been able to test the Schedule module with multiple attendees since the model is not able to correctly parse multiple attendees. But it should still work in the future if we support it since I remove/update the resolved entries in `to` and recursively call the `parse_processed` function to continue resolving any attendees.

# Related issue
Closes #74 and Closes #73 
If you can think of any more existing functionality that should be moved to the utils folder pls move them

### CODE REVIEW

**Review by Blinningjr — APPROVED** [review]
Looks good!

File: lib/utils/contacts.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+
+def get_emails(names):
+    """
+    Returns a dictionary containing three key-value pairs given a string list
+    containing names or emails. The key-value pairs reflects the results of
+    trying to resolve the names with known contacts. If one of the names is a
+    valid email it will be added to the resulting email list directly.
+
+    Returns dictrionary with the following key-value pairs:
```
Comment (viktorfrom) [inline lib/utils/contacts.py:15]: Just some typos here, "dictirionary", "wich", "touples"

**Review by markhakansson — APPROVED** [review]
good job

File: lib/utils/contacts.py (hunk 1 above)
Comment (hugowangler) [reply lib/utils/contacts.py:15]: whoopsie, will fix

**Pull request merged**